homeassistant/components/zwave_js/__init__.py

```diff
@@ -251,21 +258,30 @@ def async_on_notification(notification: Notification) -> None:
             },
         )
 
+    entry_hass_data: dict = {}
+    hass.data[DOMAIN][entry.entry_id] = entry_hass_data
     # connect and throw error if connection failed
     try:
         async with timeout(CONNECT_TIMEOUT):
             await client.connect()
+    except InvalidServerVersion as err:
+        if not entry_hass_data.get(DATA_INVALID_SERVER_VERSION_LOGGED):
+            LOGGER.error("Invalid server version: %s", err)
+            entry_hass_data[DATA_INVALID_SERVER_VERSION_LOGGED] = True
+        if use_addon:
+            async_ensure_addon_updated(hass)
+        raise ConfigEntryNotReady from err
     except (asyncio.TimeoutError, BaseZwaveJSServerError) as err:
-        LOGGER.error("Failed to connect: %s", err)
+        if not entry_hass_data.get(DATA_CONNECT_FAILED_LOGGED):
+            LOGGER.error("Failed to connect: %s", err)
+            entry_hass_data[DATA_CONNECT_FAILED_LOGGED] = True
         raise ConfigEntryNotReady from err
     else:
         LOGGER.info("Connected to Zwave JS Server")
```

Comment:
When we get here, we should set `DATA_CONNECT_FAILED_LOGGED` and `DATA_INVALID_SERVER_VERSION_LOGGED` to `False` to ensure that if it breaks after a reload, we're logging. 
```suggestion
        LOGGER.info("Connected to Zwave JS Server")
        entry_hass_data[DATA_CONNECT_FAILED_LOGGED] = False
        entry_hass_data[DATA_INVALID_SERVER_VERSION_LOGGED] = False        
```